cli/pkg/github/git.go

```diff
@@ -0,0 +1,47 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+)
+
+func PublicKey() (*ssh.PublicKeys, error) {
+	var publicKey *ssh.PublicKeys
+	sshPath := os.Getenv("HOME") + "/.ssh/id_ed25519"
+	sshKey, _ := os.ReadFile(sshPath)
+	publicKey, err := ssh.NewPublicKeys("git", sshKey, "")
+	if err != nil {
+		return nil, err
+	}
+	return publicKey, err
+}
+
+func CloneRepo(url string) (string, error) {
+	repoName := strings.TrimSuffix(url[strings.LastIndex(url, "/")+1:], ".git")
+	fmt.Printf("Cloning Repo URL: {%s}\n", url)
+	if strings.HasPrefix(url, "git@github.com") {
+		auth, keyErr := PublicKey()
+		if keyErr != nil {
+			return repoName, keyErr
+		}
```

Comment:
This is not sufficient for git authentication. There are multiple ways to authenticate with git, in my case I do not use SSH, and my keys are in the macOS keychain (running `git config credential.helper` outputs `osxkeychain`).

I'm not convinced `go-git` fully supports the various possible default credential sources. Relevant issues:
- https://github.com/go-git/go-git/issues/490
- https://github.com/src-d/go-git/issues/1101

I'll let you assess if you can find a way to reliably populate credentials within the library. If not, I would suggest adding a helper to call the "real" `git` command to get the credentials – based on a quick search, it appears this command does the job (but please check if that's correct): `echo url=https://github.com/rilldata/rill-developer.git | git credential fill`

Reply:
If we anyway need to rely on `git` being available on the system, you could argue we don't need `go-git` in the first place (could just invoke `git` as a subprocess). 

However, let's see if we can make this work. `go-git` will be nice for extracting info about the current repo, and we might be able to re-use some of this code for cloning repos server-side (where git won't be available).